File: core/src/test/java/ru/pocketbyte/locolaser/testutils/mock/MockResourceFile.java
package ru.pocketbyte.locolaser.testutils.mock;

import ru.pocketbyte.locolaser.config.WritingConfig;
import ru.pocketbyte.locolaser.resource.entity.ResMap;
import ru.pocketbyte.locolaser.resource.file.ResourceFile;

import java.io.IOException;

public class MockResourceFile implements ResourceFile {

    public ResMap mMap;

    public MockResourceFile(ResMap map) {
        mMap = map;
    }

    @Override
    public ResMap read() {
        return mMap;
    }

    @Override
    public void write(ResMap resMap, WritingConfig writingConfig) throws IOException {
        mMap = resMap;
    }
}

File: core/src/test/java/ru/pocketbyte/locolaser/testutils/mock/MockResourceFile.kt
package ru.pocketbyte.locolaser.testutils.mock

import ru.pocketbyte.locolaser.config.ExtraParams
import ru.pocketbyte.locolaser.resource.entity.ResMap
import ru.pocketbyte.locolaser.resource.file.ResourceFile
import ru.pocketbyte.locolaser.resource.formatting.FormattingType
import ru.pocketbyte.locolaser.resource.formatting.NoFormattingType

import java.io.IOException

class MockResourceFile(
        var map: ResMap?
) : ResourceFile {

    override val formattingType = NoFormattingType

    override fun read(extraParams: ExtraParams?): ResMap? {
        return map
    }

    @Throws(IOException::class)
    override fun write(resMap: ResMap, extraParams: ExtraParams?) {
        map = resMap
    }
}
